spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -256,4 +268,23 @@ public String toString() {
         runtimeFilterExpressions,
         caseSensitive());
   }
+
+  @Override
+  public CustomTaskMetric[] reportDriverMetrics() {
+    List<CustomTaskMetric> customTaskMetrics = Lists.newArrayList();
+    MetricsReport metricsReport = sparkReadMetricReporter.getMetricsReport();
+    ScanReport scanReport = (ScanReport) metricsReport;
```

Comment:
rather than casting here I think what would be better is to provide the necessary type of metrics report in the `SparkReadMetricsReporter`? This is because `report(..)` doesn't guarantee that you will only get a `ScanReport`